### PR TITLE
Added dynamic data and visualization of neural network test predictions.

### PR DESCRIPTION
**Pull request by cmhhelgeson**
Fixes issue #2 

Added a nivo scatterplot that dynamically generates data based on the number of points specified by the user. Adjusted aome styling within the DemoMLPTraining component to facilitate this. 

Potential Issues:

Improper styling on some Browsers (currently only testing on latest version of Chrome)
Slow or nonfunctional Web Worker performance (possibly browser-dependent, defined inline)
Hitching when changing network parameters (high number of data points or epochs may stall the main process)

### CODE REVIEW

File: demo/src/demos/MLP/index.ts
Comment (trekhleb) [inline demo/src/demos/MLP/index.ts:1]: The approach with `index.ts` files is good, however, in this repository, it is not being used. Let's remove it.

File: demo/src/main.tsx [hunk 1]
```diff
@@ -1,7 +1,7 @@
 import React from 'react'
 import ReactDOM from 'react-dom/client'
 
-import { App } from './app.tsx'
+import { App } from './App';
```
Comment (trekhleb) [inline demo/src/main.tsx]: Let's also keep the naming conventions as is. Renaming the components won't add too much value and it also violates the current conventions in the repo

File: demo/src/demos/demo-mlp-training.tsx
Comment (trekhleb) [inline demo/src/demos/demo-mlp-training.tsx:1]: Let's not delete existing demos as well. You can just add an additional demo tab/component instead. For example `demo/src/demos/demo-mlp-moon-data-training.tsx`

File: demo/public/circleWorker.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+      labels
+    }
+  }
+
+  //Generate data
+  const numDataPoints = event.data;
+  let result = generateCircleData(numDataPoints);
+
+  //Shuffle result
+  /*for (let i = result.data.length - 1; i > 0; i--) {
```
Comment (trekhleb) [inline demo/public/circleWorker.js]: Let's delete the unused code

File: demo/public/circleWorker.js
Comment (trekhleb) [inline demo/public/circleWorker.js:1]: I see that we have two `circleWorker` files in this pull request. Can we avoid having two copies of the same code in favor of just one?

File: demo/src/components/CaptionBlock.tsx
Comment (trekhleb) [inline demo/src/components/CaptionBlock.tsx:1]: Both naming conventions `CaptionBlock.tsx` and `caption-block.tsx` are good. However, we should be consistent with the naming convention that was used in this repo before and go with the `caption-block.tsx`-like naming for all new components

File: demo/src/components/demos.tsx [hunk 3]
```diff
@@ -6,7 +6,7 @@ import { H1 } from './h1'
 import { DemoMLP } from '../demos/demo-mlp'
 import { DemoNeuron } from '../demos/demo-neuron'
 import { DemoValue } from '../demos/demo-value'
-import { DemoMLPTraining } from '../demos/demo-mlp-training'
+import { DemoMLPTraining } from '../demos/MLP/demo-mlp-training'
```
Comment (trekhleb) [inline demo/src/components/demos.tsx]: You don't need to restructure or change the existing demos. Instead you may just import here your new demo, like `import { DemoMLPMoonDataTraining } from '../demos/demo-mlp-moon-data-training'`

File: demo/src/demos/MLP/dataUtils.ts [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+  data: number[][];
+  labels: number[];
+}
+
+interface DataValues {
+  dataValues: Value[][],
+  labelValues: Value[],
+}
+
+export const generateCircleData = (
```
Comment (trekhleb) [inline demo/src/demos/MLP/dataUtils.ts]: We have 3 places with copies of similar code that generates generateCircleData (`circleWorker.js`, `circleWorker.ts`, `dataUtils.ts`). Can we make these data function reusable and have them in one place only (to have a single source of truth)?

**Review by trekhleb — COMMENTED** [review]
Thanks for the PR! I left several comments in the code. Feel free to address them once you have time

**Comment by cmhhelgeson** [comment]
> Thanks for the PR! I left several comments in the code. Feel free to address them once you have time

Thanks for taking a look! I'm out for most of today, but I should be able to fix the outstanding issues tonight.

**Comment by trekhleb** [comment]
Thanks for the PR @cmhhelgeson!

**Pull request merged**